src/hooks/useUser/useUser.tsx

```diff
@@ -0,0 +1,109 @@
+import { useQuery } from '@tanstack/react-query'
+import { useMemo } from 'react'
+import { v4 as uuidv4 } from 'uuid'
+
+import { getExpoToken } from '@/context/WalletProvider/MobileWallet/mobileMessageHandlers'
+import { useFeatureFlag } from '@/hooks/useFeatureFlag/useFeatureFlag'
+import { isMobile } from '@/lib/globals'
+import { getOrCreateUser, registerDevice } from '@/lib/user/api'
+import type { User } from '@/lib/user/types'
+import { DeviceType } from '@/lib/user/types'
+import { selectWalletEnabledAccountIds } from '@/state/slices/common-selectors'
+import { useAppSelector } from '@/state/store'
+
+type UseUserData = {
+  user: User | null
+  expoToken: string | null | undefined
+  isLoadingUser: boolean
+  isLoadingDevice: boolean
+  isLoading: boolean
+  error: Error | null
+  refetchUser: () => void
+  refetchDevice: () => void
+}
+
+export const useUser = (): UseUserData => {
+  const walletEnabledAccountIds = useAppSelector(selectWalletEnabledAccountIds)
+  const isWebservicesEnabled = useFeatureFlag('Webservices')
+
+  const {
+    data: mobileExpoToken,
+    isLoading: isLoadingExpoToken,
+    error: expoTokenError,
+  } = useQuery({
+    queryKey: ['expoToken'],
+    queryFn: getExpoToken,
+    enabled: isMobile && isWebservicesEnabled,
+    gcTime: Infinity,
+    staleTime: Infinity,
+  })
+
+  const {
+    data: userData,
+    isLoading: isLoadingUser,
+    error: userError,
+    refetch: refetchUser,
+  } = useQuery({
+    queryKey: ['user', walletEnabledAccountIds],
+    queryFn: () => getOrCreateUser({ accountIds: walletEnabledAccountIds }),
+    enabled: walletEnabledAccountIds.length > 0 && isWebservicesEnabled,
+    staleTime: Infinity,
+    gcTime: Infinity,
+  })
+
+  const {
+    data: deviceData,
+    isLoading: isLoadingDevice,
+    error: deviceError,
+    refetch: refetchDevice,
+  } = useQuery({
```

Comment:
q: is this being a query vs. a mutation intended here? i.e it calls `registerDevice()` so feels like it should be a mutation?

If the intent is `getOrCreate()` pattern (which looks like it's the case here) then we could use such terminology for `registerDevice()` to make this clear